File: 0456-132-pattern/0456-132-pattern.cpp
class Solution {
public:
    bool find132pattern(vector<int>& nums) {
      int third_value = INT_MIN;
      stack <int> last;
      for(int i = nums.size() - 1;i >= 0;i--){
       if(nums[i] < third_value)return true;
       while(!last.empty() && last.top() < nums[i])
           third_value = last.top() , last.pop();
       last.push(nums[i]);   
      }  
      return false;  
    }
};

File: 0456-132-pattern/0456-132-pattern.rb
# @param {Integer[]} nums
# @return {Boolean}
def find132pattern(nums)
    stack = []
    minimum = -Float::INFINITY
    nums = nums.reverse
    nums.each_with_index do |val,index|
        return true if minimum > val
        minimum = stack.pop while !stack.empty? and stack[-1] < val
        stack.push(val)
    end
    false
end